packages/interactive-messages/src/http-handler.ts

```diff
@@ -40,13 +38,13 @@ export function createHTTPHandler(adapter) {
   /**
    * Parses raw bodies of requests
    *
-   * @param {string} body - Raw body of request
-   * @returns {Object} Parsed body of the request
+   * @param body - Raw body of request
+   * @returns Parsed body of the request
    */
-  function parseBody(body) {
+  function parseBody(body: string): any {
```

Comment:
can we make some better assumptions around the type here? maybe at least `Record<string, any>`?

as long as know `payload` is always a plain object or undefined (not an array/number/string).

Reply:
@aoberoi 👀💭☝️❓ (I'm impartial on this)

Reply:
FYI too the reason being we cast elsewhere in the file where we really shouldn't have to. also `any` is best avoided really, especially with the introduction of `unknown` these days. it is better and safer to be more specific about the type or at least use `unknown` and do stricter checks before using the value.

Reply:
This function returns the result of `JSON.parse()`, [which is an `any`](https://github.com/microsoft/TypeScript/blob/d982014d733445f478b5cad8a938d2d2a56f1080/lib/lib.es5.d.ts#L1053-L1059). Using any other type would require a cast here, so I don't agree with the motivation of removing casts. I think the cast being made local to the call site of `parseBody()` has the benefit of keeping that knowledge of what the input was close to the location of the cast.

I think the TypeScript team wasn't just lazy and did this for good reason. `any` is actually a better model for an arbitrary JavaScript object than the other options. We had a similar discussion about using `object` to describe this type of object. Specifically compared to [`unknown`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type), the following properties make it a less correct model for this value:

```
// keyof unknown

type T40 = keyof any;  // string | number | symbol
type T41 = keyof unknown;  // never
```

The keys of an arbitrary JavaScript object are actually `string | number | symbol`. That may not be true for every kind value that `JSON.parse()` produces, but since `string` is a subset, that is still true. Whereas `never` is actually untrue.

Reply:
I understand exactly that JSON parse is any and it definitely should be. 

My question was if you can make any assumptions about the payload. If it really can be anything, the answer is no. But I wasn't sure if you knew that payloads here have a known structure.

Reply:
🤔 If we wanted to produce a type that could describe the structure any more specifically than "the type that `JSON.parse()` returns", I think it would need to model the kinds of payloads Slack might deliver over the Interactive Components part of the platform. That's a worthy idea! In fact, [we've done much of that work over in Bolt](https://github.com/slackapi/bolt/tree/master/src/types/actions). IMHO, the `Record<string, any>` model doesn't add much value over `any` - in fact I think it's not really what the `Record` type is meant for. The one thing you'd get is the ability to index the value with strings, [but the `any` type already allows you to do this](http://www.typescriptlang.org/play/#code/CYUwxgNghgTiAEYD2A7AzgF3gDwFzyhQE8BuAKDOXSQhADoIkBzACmzoDMkkBKcs0JFgIqmeEXwAlcEhjAAPJhgBLFEwA0BYgD5+omvUasidAEaw+FIA). Side note: if I wanted a way to do this without `any`, I'd likely reach for an interface with an index type instead of `Record`, which I think communicates a very different intention.

So, can we give this a type that actually adds value? Tactically, we've made the choice to extract common Slack types that are used across many of our packages in `@slack/types` so that third party packages (ones that Slack doesn't maintain) may also depend on them. We haven't done much to promote that yet, because it's still early days. We probably want to do some more testing before we start encouraging that type of use so that certain usages don't regress (there's not a lot of precedent for how to do semver for the types of a package). Long story short: I think there's an opportunity to move many of the types currently in Bolt into `@slack/types`, and then to use them in this package. That would be cool!

Now that this PR has landed, if we want to continue this discussion, we should create a new issue for it. I'd be happy to hear more.